crates/autopilot/src/lib.rs

```diff
@@ -607,11 +607,31 @@ pub async fn main(args: arguments::Arguments) -> ! {
         LimitOrderMetrics {
             quoting_age: limit_order_age,
             validity_age: limit_order_age * SURPLUS_FEE_EXPIRATION_FACTOR.into(),
-            database: db,
+            database: db.clone(),
         }
         .spawn();
     }
 
-    let result = serve_metrics.await;
-    panic!("serve_metrics exited {result:?}");
+    if args.enable_colocation {
+        if args.drivers.is_empty() {
+            panic!("colocation is enabled but no drivers are configured");
+        }
+        let run = run_loop::RunLoop {
+            solvable_orders_cache,
+            database: db,
+            drivers: args
+                .drivers
+                .into_iter()
+                .map(driver_api::Driver::new)
+                .collect(),
+            current_block: current_block_stream,
+            web3,
+            network_block_interval: _network_time_between_blocks,
+        };
+        run.run_forever().await;
+        panic!("run loop exited");
```

Comment:
Slightly slightly better
```suggestion
        unreachable!("run loop exited");
```